merkle/tree_hasher.go

```diff
@@ -14,11 +14,24 @@
 
 package merkle
 
-// TreeHasher is the interface that the previous tree hasher struct implemented.
+// TreeHasher computes generic merkle trees.
```

Comment:
how's "provides the hash functions needed to compute both chron and sparse merkle trees."